src/lib/dialog/dialog-ref.ts

```diff
@@ -14,10 +14,18 @@ export class MdDialogRef<T> {
   /** The instance of component opened into the dialog. */
   componentInstance: T;
 
+  /** Expose overlay backdrop click event */
+  backdropClicked: Observable<void>;
+
+  /** Subject for notifying the user that esc key way pressed */
+  escapePressed = new Subject<void>();
```

Comment:
I'm not sure that we need observables for `escapePressed`, `backdropClicked` and `afterClosed` since the API is already becoming pretty dense. I'd rather we kept everything to `afterClosed` and emitted an object that looks something like `{ result: <user-provided value>, reason: 'backdrop' | 'keypress' | 'api' }`.